Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -598,6 +598,7 @@ extension RouteController: Router {
                 strongSelf._routeProgress = RouteProgress(route: route, options: routeOptions, legIndex: 0)
                 strongSelf._routeProgress.currentLegProgress.stepIndex = 0
                 strongSelf.announce(reroute: route, at: location, proactive: false)
+                strongSelf.indexedRouteResponse = .init(routeResponse: response, routeIndex: 0)
```

Comment:
This is correct for now. #3345 will change the behavior of the method that this closure is passed into, so that the closest matching route will no longer be guaranteed to be at index 0; instead, the closure will receive the index of the closest matching route.